src/components/structures/RoomDirectory.js

```diff
@@ -278,7 +267,7 @@ module.exports = React.createClass({
         this.filterTimeout = setTimeout(() => {
             this.filterTimeout = null;
             this.refreshRoomList();
-        }, 300);
+        }, 700);
```

Comment:
700ms still feels pretty responsive, and will help further reduce the number of hits to the API